KMeans.java

```diff
@@ -0,0 +1,126 @@
+import java.io.*;
+import java.util.*;
+import java.lang.Math; 
+
+public class KMeans {
+
+	private Vector midpoint (List<Vector> listofvectors) { //computes + returns a midpoint Vector from a given list of Vectors
+		
+		List<Double> mp_coordinates = new ArrayList<Double>();
+		double number_of_vectors = listofvectors.size();
```

Comment:
`number_of_vectors` --> `numVectors`.  And this should be an `int` not a `double`.  Also, you need error checks to make sure that all the vectors in the given list are of the same length.